src/Button.js

```diff
@@ -13,12 +13,26 @@ class Button extends Component {
   }
 
   componentDidMount() {
-    const { tooltip, tooltipOptions } = this.props;
-    if (
-      typeof $ !== 'undefined' &&
-      (typeof tooltip !== 'undefined' || typeof tooltipOptions !== 'undefined')
-    ) {
-      $(this._btnEl).tooltip(tooltipOptions);
+    if (!M) return;
+
+    const { tooltip, tooltipOptions, fab } = this.props;
+    if (tooltip) {
+      this.tooltipInstance = M.Tooltip.init(this._btnEl, tooltipOptions);
```

Comment:
We can reuse this `tooltipInstance` on L24 and then you would only need to destroy this instance on unmount instead of having to check, see L30

Reply:
So I basically would have one this.instance and only one destroy? @alextrastero

Reply:
Yeah, give it a go

Reply:
Done!